tests/system/src/test/java/test/javafx/scene/web/CSSFilterTest.java

```diff
@@ -0,0 +1,185 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.web;
+
+import java.util.concurrent.CountDownLatch;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.image.PixelReader;
+import javafx.scene.image.WritableImage;
+import javafx.scene.paint.Color;
+import javafx.scene.web.WebView;
+import javafx.stage.Stage;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import test.util.Util;
+
+import static javafx.concurrent.Worker.State.SUCCEEDED;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class CSSFilterTest {
+    private static final CountDownLatch launchLatch = new CountDownLatch(1);
+
+    // Maintain one application instance
+    static CSSFilterTestApp cssFilterTestApp;
+
+    private WebView webView;
+
+    public static class CSSFilterTestApp extends Application {
+        Stage primaryStage = null;
+
+        @Override
+        public void init() {
+            CSSFilterTest.cssFilterTestApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            Platform.setImplicitExit(false);
+            this.primaryStage = primaryStage;
+            launchLatch.countDown();
+        }
+    }
+
+    private static String colorToString(Color c) {
+        int r = (int)(c.getRed() * 255.0);
+        int g = (int)(c.getGreen() * 255.0);
+        int b = (int)(c.getBlue() * 255.0);
+        int a = (int)(c.getOpacity() * 255.0);
+        return "rgba(" + r + "," + g + "," + b + "," + a + ")";
+    }
+
+    private void assertColorEquals(String msg, Color expected, Color actual, double delta) {
+        if (!testColorEquals(expected, actual, delta)) {
+            fail(msg + " expected:" + colorToString(expected)
+                    + " but was:" + colorToString(actual));
+        }
+    }
+
+    private void assertColorNotEquals(String msg, Color notExpected, Color actual, double delta) {
+        double deltaRed = Math.abs(notExpected.getRed() - actual.getRed());
+        double deltaGreen = Math.abs(notExpected.getGreen() - actual.getGreen());
+        double deltaBlue = Math.abs(notExpected.getBlue() - actual.getBlue());
+        double deltaOpacity = Math.abs(notExpected.getOpacity() - actual.getOpacity());
+        if (deltaRed < delta && deltaGreen < delta && deltaBlue < delta && deltaOpacity < delta) {
```

Comment:
testColorEquals can be reused here.
`if (testColorEquals(expected, actual, delta))`